### PR TITLE
Feature/PXS-6168 add premium theme

### PR DESCRIPTION
**Pull request by shruti91sri**
Added a dropdown to toggle between themes

https://github.com/user-attachments/assets/afc55dca-e5fd-405f-9b60-388fb8484985

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "@backbase/internal-accessgroup-shared-data-access": "4.1.0",
     "@backbase/select-context-widget-ang": "8.2.0",
     "@backbase/transactions-http-ang": "^4.0.0",
-    "@backbase/ui-ang": "^11.0.0",
+    "@backbase/ui-ang": "^11.11.0",
```
Comment (ycewe) [inline package.json:48]: I didn't see a change in the `package-lock.json`. Is it really updated?

File: apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+selector: 'app-theme-switcher',
+templateUrl: './theme-switcher.component.html',
+})
+export class ThemeExporterComponent{
+  @Input()
+  theme = ThemeService.THEMES[0];
+
+  themes = ThemeService.THEMES;
+
+  onSelect(theme: any) {
```
Comment (ycewe) [inline apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts:14]: Maybe `theme: ThemeService.THEMES` or string?

File: apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+@Component({
+selector: 'app-theme-switcher',
+templateUrl: './theme-switcher.component.html',
+})
+export class ThemeExporterComponent{
+  @Input()
+  theme = ThemeService.THEMES[0];
+
+  themes = ThemeService.THEMES;
```
Comment (ycewe) [inline apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts:12]: Is this used?

File: apps/golden-sample-app/src/app/theme-switcher/theme-service.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class ThemeService {
+  public static THEMES = [
+    'theme-default',
+    'theme-premium',
+  ];
+
+  public static setTheme(theme: any):void {
```
Comment (ycewe) [inline apps/golden-sample-app/src/app/theme-switcher/theme-service.ts:10]: Do you need these to be `static`? How about following the documentation in https://designsystem.backbase.com/theme/multi-theming-on-web#2-implementing-the-theme-manager? If you want to it a different way, you may need to update the documentation also.

File: apps/golden-sample-app/project.json [hunk 5]
```diff
@@ -30,7 +30,19 @@
             "output": "assets/transactions-journey"
           }
         ],
-        "styles": ["apps/golden-sample-app/src/styles.scss"],
+        "styles": [
+          "apps/golden-sample-app/src/styles.scss",
```
[the remark this answers is not in the document]
Comment (shruti91sri) [reply apps/golden-sample-app/project.json:34]: forgot to remove

File: apps/golden-sample-app/src/app/app.module.ts [hunk 6]
```diff
@@ -47,6 +47,8 @@ import { TrackerModule } from '@backbase/foundation-ang/observability';
 import { UserContextInterceptor } from './user-context/user-context.interceptor';
 import { ActivityMonitorModule } from './auth/activity-monitor';
 import packageInfo from 'package-json';
+import { ThemeExporterModule } from './theme-switcher/theme-switcher.component.module';
```
[the remark this answers is not in the document]
Comment (shruti91sri) [reply apps/golden-sample-app/src/app/app.module.ts:50]: Copy paste error

File: apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts (hunk 3 above)
Comment (shruti91sri) [reply apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts:12]: `themes` is being consumed by `[options]` of the `dropdown-menu-ui` in the HTML

File: apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts (hunk 2 above)
Comment (shruti91sri) [reply apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts:14]: Well i set it to as `onselect($event)` is passed as parameter 
i have to add `object` as well as it was throwing error that `$event` can be `string or object`  and then checking in the function if it's a `string` then the logic follows 
```onSelect(theme: string | object) {
    if(typeof theme === 'string'){
      ThemeService.setTheme(theme);
      this.theme = theme;
    }
  }

File: apps/golden-sample-app/src/app/theme-switcher/theme-service.ts (hunk 4 above)
Comment (shruti91sri) [reply apps/golden-sample-app/src/app/theme-switcher/theme-service.ts:10]: updated

File: apps/golden-sample-app/project.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (ycewe) [reply apps/golden-sample-app/project.json:34]: Did you try changing https://github.com/Backbase/golden-sample-app/pull/238/files#diff-3030dde92fa4cb31c656598b1b4d2fa4ffb0624031ece99e6bbb6637d4cafc7bR38? To `inject: true`? You can probably refer to our showcase-app.

File: apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts (hunk 2 above)
Comment (ycewe) [reply apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts]: Yes, this sounds like a better practice. `string` or `Event` type maybe?

File: apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+@Component({
+  selector: 'app-theme-switcher',
+  templateUrl: './theme-switcher.component.html',
+})
+export class ThemeSwitcherComponent {
+  constructor(private themeManagerService: ThemeManagerService) {}
+
+  @Input() theme = ThemeManagerService.THEMES[0];
+
+  themes = ThemeManagerService.THEMES;
```
Comment (ycewe) [inline apps/golden-sample-app/src/app/theme-switcher/theme-switcher.component.ts:13]: If this is not used anywhere, and only in the HTML, `protected` might be better. Same for `onSelect`.

File: apps/golden-sample-app/src/themes/theme-premium/main.scss [hunk 8]
```diff
@@ -0,0 +1,6 @@
+@import "./variables";
+
+@import '@backbase/ui-ang/scss/main';
+
+// This will be used by apps when they request for the premium theme
+// @import "@backbase/ui-ang/scss/main.scss";
```
Comment (ycewe) [inline apps/golden-sample-app/src/themes/theme-premium/main.scss:6]: You can probably remove these comment

**Review by ycewe — COMMENTED** [review]
Small comments left. LGTM

File: apps/golden-sample-app/project.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (shruti91sri) [reply apps/golden-sample-app/project.json:34]: i did that in my inital commits https://github.com/Backbase/golden-sample-app/pull/238/commits/b448f2d45ffdbb43a9dc1ea4860218e81828cd3f#diff-3030dde92fa4cb31c656598b1b4d2fa4ffb0624031ece99e6bbb6637d4cafc7bR38

[the remark this answers is not in the document]
Comment (shruti91sri) [reply apps/golden-sample-app/project.json]: it worked thanks

**Review by ycewe — APPROVED** [review]

**Review by ankypant — APPROVED** [review]

**Pull request merged**